core/src/mindustry/mod/Scripts.java

```diff
@@ -81,8 +94,30 @@ private boolean run(String script, String file){
         }
     }
 
+    private String fillWrapper(Fi file){
+        return wrapper.replace("$SCRIPT_NAME$", loadedMod.name + "/" + file.nameWithoutExtension())
+            .replace("$CODE$", file.readString())
+            .replace("$MOD_NAME$", loadedMod.name);
+    }
+
     @Override
     public void dispose(){
         Context.exit();
     }
+
+    private class ScriptModuleProvider extends UrlModuleSourceProvider{
+        public ScriptModuleProvider(){
+            super(null, null);
+        }
+
+        @Override
+        public ModuleSource loadSource(String moduleId, Scriptable paths, Object validator) throws IOException, URISyntaxException{
+            if(loadedMod == null) return null;
+            Fi module = loadedMod.root.child("scripts").child(moduleId + ".js");
```

Comment:
This could be more powerful if it allowed for cross-mod script loading. `A` could depend on `B`, and `A` could include `B`'s `utilities.js` by calling `require("B/utilities")`.